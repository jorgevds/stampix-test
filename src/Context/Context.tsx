import { createContext } from "react";

export const LanguageContext = createContext<{
  language?: any;
  changeLanguage: any;
}>({
  language: "en",
  changeLanguage: () => {},
});
