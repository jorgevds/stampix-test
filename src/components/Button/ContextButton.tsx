import React, { useContext } from "react";
import { LanguageContext } from "../../Context/Context";

interface ContextButtonProps {
  children?: React.ReactNode;
  value: string;
  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

const ContextButton: React.FC<ContextButtonProps> = ({ children, value }) => {
  const { language, changeLanguage } = useContext(LanguageContext);

  const onButtonClick = (
    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => {
    changeLanguage(e.currentTarget.value);
  };

  return (
    <button onClick={onButtonClick} value={value} id={language}>
      {children}
    </button>
  );
};

export default ContextButton;
