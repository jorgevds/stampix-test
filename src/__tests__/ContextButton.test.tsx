import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import ContextButton from "../components/Button/ContextButton";
import { LanguageContext } from "../Context/Context";

configure({ adapter: new Adapter() });
describe("Context button works", () => {
  const context = shallow(
    <LanguageContext.Provider
      value={{ language: "en", changeLanguage: () => {} }}
    />
  );

  it("Context Provider should render correctly", () =>
    expect(context).toMatchSnapshot());

  it("ContextButton triggers the setState hook called changeLanguage", () => {
    const testState: any = { value: "nl" };
    const changeLanguage = jest.fn();

    const button = shallow(
      <ContextButton value="nl" onClick={changeLanguage} />
    );

    const handleClick = jest.spyOn(React, "useState");
    handleClick.mockImplementation(() => [testState, changeLanguage]);

    expect(button.props().value).toEqual("nl");

    button.simulate("click", {
      currentTarget: { name: "value", value: "nl" },
    });

    expect(changeLanguage).toBeTruthy();
  });
});
