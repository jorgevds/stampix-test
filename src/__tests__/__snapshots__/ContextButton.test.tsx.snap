// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context button works Context Provider should render correctly 1`] = `ShallowWrapper {}`;
